CharlieBackend.Api/Controllers/CoursesController.cs

```diff
@@ -4,37 +4,39 @@
 using CharlieBackend.Core.Models.Course;
 using Microsoft.AspNetCore.Authorization;
 using CharlieBackend.Business.Services.Interfaces;
-
+using CharlieBackend.Core.DTO.Course;
+using AutoMapper;
 
 namespace CharlieBackend.Api.Controllers
 {
     [Route("api/courses")]
     [ApiController]
     public class CoursesController : ControllerBase
     {
-        #region
+      
         private readonly ICourseService _coursesService;
-        #endregion
+        private readonly IMapper _mapper;
 
-        public CoursesController(ICourseService coursesService)
+        public CoursesController(ICourseService coursesService, IMapper mapper)
         {
             _coursesService = coursesService;
+            _mapper = mapper;
         }
 
         [Authorize(Roles = "4")]
         [HttpPost]
-        public async Task<ActionResult> PostCourse(CreateCourseModel courseModel)
+        public async Task<ActionResult> PostCourse(CreateCourseDto courseDto)
         {
             if (!ModelState.IsValid) return BadRequest();
 
-            var isCourseNameTaken = await _coursesService.IsCourseNameTakenAsync(courseModel.Name);
+            var isCourseNameTaken = await _coursesService.IsCourseNameTakenAsync(courseDto.Name);
 
             if (isCourseNameTaken)
             {
                 return StatusCode(409, "Course already exists!");
             }
 
-            var createdCourse = await _coursesService.CreateCourseAsync(courseModel);
+            var createdCourse = await _coursesService.CreateCourseAsync(_mapper.Map<CreateCourseModel>(courseDto));
```

Comment:
You don't need to use mapper in controller. As suggested in previous comments, please leave only dto or model